### PR TITLE
fix: consider pods with state succeeded also valid

### LINKED ISSUE #2089: Execution fails for packages running one off containers in Kubernetes
### What's your CLI version?

0.86.11

### Description & steps to reproduce

After setting up kurtosis in kubernetes, running the below command 
```
kurtosis run github.com/hugobyte/polkadot-kurtosis-package '{ "chain_type": "localnet", "relaychain": { "name": "rococo-local", "nodes": [ { "name": "alice", "node_type": "validator", "prometheus": false }, { "name": "bob", "node_type": "full", "prometheus": true } ] }, "parachains": [ { "name":"frequency", "nodes": [ { "name": "alice", "node_type": "validator", "prometheus": false } ] } ], "explorer": true }'
```

fails with the error
```
DEBU[2024-01-26T04:39:57Z][api_container_service.go:runStarlark] Received response line from Starlark runner: 'error:{execution_error:{error_message:"An error occurred executing instruction (number 8) at github.com/hugobyte/polkadot-kurtosis-package/package_io/build-spec.star[19:31]:\nadd_service(name=\"para-slot-registration\", config=ServiceConfig(image=\"hugobyte/parachain-node-modules\", files={\"/app\": \"configs\", \"/build\": \"javascript\"}, entrypoint=[\"/bin/sh\"]))\n --- at /home/circleci/project/core/server/api_container/server/startosis_engine/startosis_executor.go:157 (sendErrorAndFail) ---\nCaused by: Unexpected error occurred starting service 'para-slot-registration'\n --- at /home/circleci/project/core/server/api_container/server/startosis_engine/kurtosis_instruction/add_service/add_service.go:163 (AddServiceCapabilities.Execute) ---\nCaused by: An error occurred creating pod 'para-slot-registration' using image 'hugobyte/parachain-node-modules'\n --- at /home/circleci/project/container-engine-lib/lib/backend_impls/kubernetes/kubernetes_kurtosis_backend/user_services_functions/start_user_services.go:421 (createStartServiceOperation.func1) ---\nCaused by: An error occurred waiting for pod 'para-slot-registration' to become available\n --- at /home/circleci/project/container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager/kubernetes_manager.go:1159 (KubernetesManager.CreatePod) ---\nCaused by: Expected state of pod 'para-slot-registration' to arrive at 'Running' but the pod instead landed in 'Succeeded' with the following state:\n>>>>>>>>>>>>>>>>>>>>>>>>>> Pod para-slot-registration <<<<<<<<<<<<<<<<<<<<<<<<<<\nContainer Statuses: - user-service-container (docker.io/hugobyte/parachain-node-modules:latest): TERMINATED with exit code 0 - \n\n-------------------- Container user-service-container Logs --------------------\n\n------------------ End Container user-service-container Logs ---------------------\n>>>>>>>>>>>>>>>>>>>>>>>> End Pod para-slot-registration <<<<<<<<<<<<<<<<<<<<<<<<<<<\n --- at /home/circleci/project/container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager/kubernetes_manager.go:1871 (KubernetesManager.waitForPodAvailability) ---"}}' 
```

which is a caused due to execution of one-off container; issue arises from https://github.com/kurtosis-tech/kurtosis/blob/main/container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager/kubernetes_manager.go#L1869

### Desired behavior

The package should complete execution without failure.

### What is the severity of this bug?

Critical; I am blocked and Kurtosis is unusable for me because of this bug.

### What area of the product does this pertain to?

CLI: the Command Line Interface

### PR DESCRIPTION
**Pull request by MuhammedIrfan**
## Description:
The Kubernetes pods that are in the state 'succeeded' should also be considered valid as one-off containers complete after execution with the state succeeded.

## Is this change user facing?
NO

## References (if applicable):
- Closes https://github.com/kurtosis-tech/kurtosis/issues/2089

### CODE REVIEW

File: container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager/kubernetes_manager.go [hunk 1]
```diff
@@ -1867,15 +1867,7 @@ func (manager *KubernetesManager) waitForPodAvailability(ctx context.Context, na
 				podStateStr,
 			)
 		case apiv1.PodSucceeded:
-			podStateStr := manager.getPodInfoBlockStr(ctx, namespaceName, pod)
```
Comment (h4ck3rk3y) [inline container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager/kubernetes_manager.go:1870]: can we move this condition to the top of the list near PodRunning and PodUnknown for consistency.

**Review by h4ck3rk3y — APPROVED** [review]
This makes sense, we have waits per port as well so if someone has a proper (non one off service); it shall wait on the port and fail if that doesn't come up.

File: container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager/kubernetes_manager.go (hunk 1 above)
Comment (MuhammedIrfan) [reply container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager/kubernetes_manager.go:1870]: Sure

**Comment by mieubrisse** [comment]
Hey @MuhammedIrfan , this pod-waiting functionality is for services (i.e. things that are supposed to start running and stay running). As far as I understand, the "succeeded" state is for a pod that starts, runs, and exits, which would be a better fit for a Kurtosis task (e.g. `run_sh` or `run_python`), which are intended to start and stop. Would you be able to use a task here? I think if we make this change, it would blur the lines between service and task

**Comment by MuhammedIrfan** [comment]
> Hey @MuhammedIrfan , this pod-waiting functionality is for services (i.e. things that are supposed to start running and stay running). As far as I understand, the "succeeded" state is for a pod that starts, runs, and exits, which would be a better fit for a Kurtosis task (e.g. `run_sh` or `run_python`), which are intended to start and stop. Would you be able to use a task here? I think if we make this change, it would blur the lines between service and task

Looks like there is a miss from our side for using add_service instead of run_sh, I will check and revert back.

**Comment by MuhammedIrfan** [comment]
> > Hey @MuhammedIrfan , this pod-waiting functionality is for services (i.e. things that are supposed to start running and stay running). As far as I understand, the "succeeded" state is for a pod that starts, runs, and exits, which would be a better fit for a Kurtosis task (e.g. `run_sh` or `run_python`), which are intended to start and stop. Would you be able to use a task here? I think if we make this change, it would blur the lines between service and task
> 
> Looks like there is a miss from our side for using add_service instead of run_sh, I will check and revert back.

This fix is no longer required for our use case. We have refactored to use run_sh.

**Comment by h4ck3rk3y** [comment]
Discussed with Irfan and the run_sh/run_python alternative was enough!

**Pull request closed without merging**